foundry/gui/orb.py

```diff
@@ -0,0 +1,362 @@
+""" Implements a UI to modify the player lives.
+
+Supported editing various characteristics of the player lives such as the
+number of starting lives, the number of lives on continues, if a death lowers
+the life count, if a 1-up mushroom increases the life count, etc.
+
+This UI uses a Redux pattern instead of MVC to increase the testability of the
+code.  If you are if you want to try to reason about Redux in terms of MVC,
+then the Redux state is the model, the Redux reducer is the Controller, and
+obviously both still have a view.
+"""
+from dataclasses import dataclass
+from enum import Enum
+
+from PySide6.QtGui import Qt
+from PySide6.QtWidgets import (
+    QBoxLayout,
+    QCheckBox,
+    QDialogButtonBox,
+    QLabel,
+    QGroupBox
+)
+
+from foundry.core.redux_store import Action, ReduxStore
+from foundry.game.File import ROM
+from foundry.gui.CustomDialog import CustomDialog
+from foundry.gui.HorizontalLine import HorizontalLine
+from foundry.smb3parse.util.code_edit_dict import CodeEditDict
+from foundry.smb3parse.util.rom import Rom
+
+
+class _UIStrings(Enum):
+    """All string values used in the UI."""
+
+    TITLE = "Orb Options"
+    INVALID_ROM_WARNING = (
+        "The selected ROM has code modifications that are"
+        " incompatible with one or more features of this window. The affected"
+        " features are visible but disabled."
+    )
+
+    MOVE_TITLE = "After touching an orb, the player can move:"
+    MOVE_TOUCH_TO_TIMER = "Until the timer is counted into the score"
+    MOVE_TIMER_TO_EXIT = "After the timer is counted until the level exits"
+    TOUCH_TITLE = "When the player touches an orb:"
+    TOUCH_GAME_TIMER_STOPS = "Stop the game timer. Allows the player to touch a stray orb without dying."
+
+
+class Actions(Enum):
+    """All user actions that can be performed."""
+
+    MOVE_TOUCH_TO_TIMER = "[Orb] MoveTouchToTimer"
+    MOVE_TIMER_TO_EXIT = "[Orb] MoveTimerToExit"
+    TOUCH_GAME_TIMER_STOPS = "[Orb] TouchGameTimerStops"
+
+
+@dataclass
+class State:
+    """Stores the current state of the UI
+
+    The initial state is read from the ROM.  The final state is written to the
+    ROM.  The intermediate values are stored in this state object while the
+    user is modifying them.
+
+    NOTE: Some of these values may be optional/None if the search for these
```

Comment:
If the state can be None, then the type hints should reflect this with the `typing.Optional` keyword.